### ISSUE TITLE
No Data available in "Event Data" Page after upgrading to 2.4.0

### ISSUE THREAD

**Issue by bayerlse** [umami-software/umami#2169]
### Describe the Bug

Hey there!

After upgrading to 2.4.0 the event data is not displayed anymore within the event data page (`<host>/websites/<id>/event-data`), despite there are entries in the `event_table` and `website_event` tables.

How could that be solved?

BR, Sebi

### Database

PostgreSQL

### Relevant log output

```shell
`event_table`


[
  {
    "event_data_id": "4a0b8dce-61ef-46b6-b5c2-7e33ecab0736",
    "website_id": "891897eb-78b1-40a7-9c96-ea60387d8379",
    "website_event_id": "46dd106e-3ac9-4c96-a9b2-7d5ba44fe353",
    "event_key": "mapInteraction",
    "string_value": "true",
    "number_value": null,
    "date_value": null,
    "data_type": 3,
    "created_at": "2023-08-01 10:09:58.662000 +00:00"
  },
  {
    "event_data_id": "c3140916-ca43-4a5d-98ea-346836fa3bd1",
    "website_id": "891897eb-78b1-40a7-9c96-ea60387d8379",
    "website_event_id": "9da66082-40fa-4ca3-9c88-cd2a4969e790",
    "event_key": "mapInteraction",
    "string_value": "true",
    "number_value": null,
    "date_value": null,
    "data_type": 3,
    "created_at": "2023-08-01 10:09:58.981000 +00:00"
  }
]


`website_event`:

```json
[
  {
    "event_id": "46dd106e-3ac9-4c96-a9b2-7d5ba44fe353",
    "website_id": "891897eb-78b1-40a7-9c96-ea60387d8379",
    "session_id": "2290fc15-3725-5eb8-a891-bebe087be111",
    "created_at": "2023-08-01 10:09:58.686000 +00:00",
    "url_path": "/",
    "url_query": null,
    "referrer_path": null,
    "referrer_query": null,
    "referrer_domain": null,
    "page_title": null,
    "event_type": 2,
    "event_name": "mapInteraction"
  },
  {
    "event_id": "9da66082-40fa-4ca3-9c88-cd2a4969e790",
    "website_id": "891897eb-78b1-40a7-9c96-ea60387d8379",
    "session_id": "2290fc15-3725-5eb8-a891-bebe087be111",
    "created_at": "2023-08-01 10:09:58.986000 +00:00",
    "url_path": "/",
    "url_query": null,
    "referrer_path": null,
    "referrer_query": null,
    "referrer_domain": null,
    "page_title": null,
    "event_type": 2,
    "event_name": "mapInteraction"
  }
]
```
```


### Which browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

Docker/Kubernetes

**Comment by garmeeh** [comment]
Hey @franciscao633 just trying out Unami at the moment (cloud version) and can see events coming through in real time data but not in "Event Data" assuming this is related? Just wanted to double check in case I had something miss configured, thanks.

**Comment by franciscao633** [comment]
Are you sending [dynamic data](https://umami.is/docs/tracker-functions) along with your events? If not, you will not see anything in this section from a normal install. It's only for those who want to see their custom event data.

**Comment by garmeeh** [comment]
Some events are sending custom data others are just simple for example `umami.track(“Login Complete”)`

I’m not self hosting, I’m using the cloud version.

**Comment by mikecao** [comment]
@garmeeh Can you send an email to support@umami.is with your account ID and we can look into it.

**Comment by garmeeh** [comment]
Hey @mikecao emailed support there earlier in the week. Would be great to know if I'm doing something wrong or not.

**Comment by raabmar** [comment]
I am not entirely sure if this should be closed? Is it working? I can't see any data at Umami Cloud. Of course there is data available just not shown in that view.

![2023-10-24_003014@2x](https://github.com/umami-software/umami/assets/15885290/2c0b722d-50a6-414e-9110-73e8435a50ef)

**Comment by franciscao633** [comment]
@raabmar can you email support@umami.is so we can verify your custom data is actually being recorded?

**Comment by uncenter** [comment]
I'm also still experiencing this issue.
![Screenshot 2023-10-24 at 16 29 04 (Arc)](https://github.com/umami-software/umami/assets/47499684/bb542239-2b0c-4933-a4ef-624d6c79b236)
![Screenshot 2023-10-24 at 16 30 21 (Arc)](https://github.com/umami-software/umami/assets/47499684/137fe17e-2ee8-4fe8-9654-042d2afdc865)

**Comment by raabmar** [comment]
> @raabmar can you email [support@umami.is](mailto:support@umami.is) so we can verify your custom data is actually being recorded?

I can send you an email about it, but similar to the next commenter the data is recorded just not showing on that page

**Comment by raabmar** [comment]
As we are on it .. when uBlock Origin is enabled in Chrome it's not showing anything at all. Completely white page. Which is odd, because why would uBlock have something blocked here, that does not make sense at all. I added this also to the support email. But also with uBlock disabled, it's not showing the data as previously mentioned

**Comment by mikecao** [comment]
@raabmar  uBlock is overly aggressive in it's blocking. It looks for words like "analytics" and "event" and if it finds it then it blocks it. Or it blocks entire domains so nothing loads. There is nothing we can do about that.

As for event data, it doesn't look like you sent any event data. Be aware that events and event data are different things.

**Comment by mikecao** [comment]
@garmeeh Sorry I missed your message. I see that you do have event data so it should be showing up now. Let me know if it doesn't.

**Comment by uncenter** [comment]
Hmm. I went to go try the page without uBlock but it was working before I even disabled it! Not sure what the issue was or what resolved it...

**Comment by raabmar** [comment]
@mikecao alright then it was my mistake and I misunderstood how it works.

I see when I add additional event data (besides the event name) it'll show up in here
![2023-10-25_003040@2x](https://github.com/umami-software/umami/assets/15885290/09dc9524-26bb-4235-bd79-186bcee2c0c5)

**Comment by uncenter** [comment]
That actually might have been my issue as well. Didn't know the event needed additional properties to show up!

**Comment by raabmar** [comment]
@uncenter glad that it was not only me - could be made a bit more clear for sure in the documentation. For new users there is no chance to grasp that just by checking the documentation or doing research. However, obviously it means the feature itself works

**Comment by uncenter** [comment]
@franciscao633 I think we've found the issue now. I think we could either display all event data (regardless of if there are additional properties) in the Event Data tab or we could try to solve this with a clearer notice on the documentation. I'm a fan of the first since it is kind of expected behavior for users.

**Comment by franciscao633** [comment]
I think those screens are built around custom event properties. If you displayed the events without properties you would end up with the same table already in the website dashboard (screenshot). We probably need to improve our documentation, or maybe find some terminology that's obvious to all users that the screen is only useful for dynamic data being passed along with the event. We're open to ideas.

![image](https://github.com/umami-software/umami/assets/31608805/4d6acd09-5075-47af-8b61-c2dedcfae44d)

**Comment by raabmar** [comment]
Just to add an opinion, I am with Umami now for a few weeks, and I can't say it is clear for users. Even showing it redundant (same information as from the dashboard) is better than not showing anything at all. Also a hint somewhere that ad blockers should be disabled would be good, with Ublock it shows a blank page. 

Why not showing two sections on the Event data (or just "Events" tab)? 
1. Events overview (if available)
2. Events details (with the additional event data if available)

You could even go one small step further, i.e. not showing the entire tab if there are no events set. Users who see screenshots of the Events tab in the documentation will ask (or complain) why they don't see the Event tab. But now those very same users ask why the Event tab is empty.

Alternatively, you could disable and grey out the Events tab and show a quick info on however that shows where events can be added.

Hope that helps!

**Comment by uncenter** [comment]
I also have Ublock Origin and I'm not getting anything blocked. Are you on Umami Cloud?

**Comment by raabmar** [comment]
> I also have Ublock Origin and I'm not getting anything blocked. Are you on Umami Cloud?

Yes Umami Cloud, I am not aware that I ever changed something in the Ublock filters. Only right now I whitelisted Umami EU
https://analytics.eu.umami.is/

**Comment by mikecao** [comment]
Fixed in v2.9.0.
